src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -364,6 +375,37 @@ internal async Task<(IReadOnlyList<T> results, string continuationToken)> Execut
             return (results, page.ContinuationToken);
         }
 
+        private void UpdateSortIndex(FhirCosmosResourceWrapper cosmosWrapper)
+        {
+            Dictionary<string, SearchParameterInfo> searchParameters = _supportedSearchParameters.Value.GetSearchParameters(cosmosWrapper.ResourceTypeName)
```

Comment:
Can we avoid recomputing this dictionary with every request?

Reply:
This is a little trickier. This should probably be pushed back up into SupportedSearchParameterDefinitionManager since the states in SearchParameterInfo are loaded from the db and change over time.

The dictionary at worst is being created on only the search params for this resource type. I'll look into this further, but the change carries more risk